# Mini Project-1
### Make informative observations over the characteristics of the data
### Observations to Focus on: Funding amounts, Type of funding 
### Simple plots and maps can be generated with the data from pandas in plotly
### Conclude your investigation into the dataset with the support of the plot.ly library for creating graphics
### Possible Variables to consider: Regional, Amounts and Classification Types
### Possible Outputs to consider: Graphics, tabular tables of statistics, perhaps a single number

In [2]:
import numpy as np
import pandas as pd
import scipy as sp
import plotly as py
import plotly.graph_objs as go
import plotly.figure_factory as ff
py.offline.init_notebook_mode(connected=True)

#### Importing the libraries that will be used in the data analysis

In [3]:
fundingdata=pd.read_csv('userssharedsdfschoolimprovement2010grants.csv')

#### Loding the data file.

In [4]:
fundingdata.head()

School Name      City State  \
0  HOGARTH KINGEEKUK MEMORIAL SCHOOL  SAVOONGA    AK   
1                    AKIACHAK SCHOOL  AKIACHAK    AK   
2                     GAMBELL SCHOOL   GAMBELL    AK   
3               BURCHELL HIGH SCHOOL   WASILLA    AK   
4                       AKIAK SCHOOL     AKIAK    AK   

                               District Name 2010/11/Award Amount  \
0              BERING STRAIT SCHOOL DISTRICT           $471014.00   
1                     YUPIIT SCHOOL DISTRICT           $520579.00   
2              BERING STRAIT SCHOOL DISTRICT           $449592.00   
3  MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT           $641184.00   
4                     YUPIIT SCHOOL DISTRICT           $399686.00   

   Model Selected                                           Location  
0  Transformation  200 MAIN ST\nSAVOONGA, AK 99769\n(63.6687, -17...  
1  Transformation  AKIACHAK 51100\nAKIACHAK, AK 99551\n(60.8911, ...  
2  Transformation  169 MAIN ST\nGAMBELL, AK 99742\n(63.7413, -171...  
3  Transformation  1775 WEST PARKS HWY\nWASILLA, AK 99654\n(61.57...  
4  Transformation     AKIAK 5227\nAKIAK, AK 99552\n(60.8879, -161.2)

#### Checking the data

In [5]:
fundingdata.tail()

School Name        City State  \
826         MOUNT HOPE HIGH SCHOOL  MOUNT HOPE    WV   
827        GEARY ELEMENTARY SCHOOL   LEFT HAND    WV   
828            TRIUMPH HIGH SCHOOL    CHEYENNE    WY   
829  HEM JUNIOR/SENIOR HIGH SCHOOL       HANNA    WY   
830          ROOSEVELT HIGH SCHOOL      CASPER    WY   

                         District Name 2010/11/Award Amount  Model Selected  \
826             FAYETTE COUNTY SCHOOLS           $342189.00  Transformation   
827               ROANE COUNTY SCHOOLS           $260000.00  Transformation   
828  LARAMIE COUNTY SCHOOL DISTRICT #1           $297300.00  Transformation   
829   CARBON COUNTY SCHOOL DISTRICT #2           $274000.00  Transformation   
830  NATRONA COUNTY SCHOOL DISTRICT #1           $431382.00  Transformation   

                                              Location  
826  110 HIGH SCHOOL DR\nMOUNT HOPE, WV 25880\n(37....  
827  9538 CLAY RD\nLEFT HAND, WV 25251\n(38.6188, -...  
828  2201 MORRIE AVE\nCHEYENNE, WY 82001\n(41.1423,...  
829    US HWY 72\nHANNA, WY 82327\n(41.8723, -106.548)  
830  140 EAST K ST\nCASPER, WY 82601\n(42.861, -106...

##### Pandas is showing 831 records. Check using describe.

In [6]:
fundingdata.describe()

School Name          City State         District Name  \
count                831           831   831                   831   
unique               821           424    50                   421   
top     NORTHWEST MIDDLE  PHILADELPHIA    CA  PHILADELPHIA CITY SD   
freq                   2            29    92                    27   

       2010/11/Award Amount  Model Selected  \
count                   757             825   
unique                  651               4   
top                   $2.00  Transformation   
freq                     17             608   

                                                 Location  
count                                                 831  
unique                                                823  
top     6941 NORTH CENTRAL ST\nPORTLAND, OR 97203\n(45...  
freq                                                    3

The count is 831. Index stops at 830 becasue Python reads from 0. The data is for 50 states.
The describe() method is currently not yeilding the statistics. This is likely becasue the Award Amounts are not integers or floats. Check Data Type.

#### Since the client is interested in the entire data and not just the data for Philadelhia school district, use the entire data.
#### First determine the data types in the filoe provided

In [7]:
fundingdata.dtypes

School Name             object
City                    object
State                   object
District Name           object
2010/11/Award Amount    object
Model Selected          object
Location                object
dtype: object

#### Since the analysis is based on funding, we need to convert Award Amount Column to Integers or Floats. The only non-numerical charecter in Amounts column is "Dollar" sign. By removing the "Dollar " sign we should be able to define floats. 

In [8]:
fundingdata['2010/11/Award Amount']= fundingdata['2010/11/Award Amount'].str.replace('$','')

In [9]:
fundingdata['2010/11/Award Amount']= fundingdata['2010/11/Award Amount'].astype(float)
fundingdata.head()


School Name      City State  \
0  HOGARTH KINGEEKUK MEMORIAL SCHOOL  SAVOONGA    AK   
1                    AKIACHAK SCHOOL  AKIACHAK    AK   
2                     GAMBELL SCHOOL   GAMBELL    AK   
3               BURCHELL HIGH SCHOOL   WASILLA    AK   
4                       AKIAK SCHOOL     AKIAK    AK   

                               District Name  2010/11/Award Amount  \
0              BERING STRAIT SCHOOL DISTRICT              471014.0   
1                     YUPIIT SCHOOL DISTRICT              520579.0   
2              BERING STRAIT SCHOOL DISTRICT              449592.0   
3  MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT              641184.0   
4                     YUPIIT SCHOOL DISTRICT              399686.0   

   Model Selected                                           Location  
0  Transformation  200 MAIN ST\nSAVOONGA, AK 99769\n(63.6687, -17...  
1  Transformation  AKIACHAK 51100\nAKIACHAK, AK 99551\n(60.8911, ...  
2  Transformation  169 MAIN ST\nGAMBELL, AK 99742\n(63.7413, -171...  
3  Transformation  1775 WEST PARKS HWY\nWASILLA, AK 99654\n(61.57...  
4  Transformation     AKIAK 5227\nAKIAK, AK 99552\n(60.8879, -161.2)

In [10]:
fundingdata.describe()

2010/11/Award Amount
count            757.000000
mean          420827.894320
std           287458.053183
min                1.000000
25%           162541.000000
50%           407794.000000
75%           671142.000000
max           997852.000000

In [11]:
fundingdata.dtypes

School Name              object
City                     object
State                    object
District Name            object
2010/11/Award Amount    float64
Model Selected           object
Location                 object
dtype: object

#### The items in Amounts column are now float objects. The describe()function above shows the statistics - including mean and Std Deviation. However the summary shows only 757 counts for funding amounts and not 831 that are needed. It is likely that some of the funding amount cells are nan values.

#### Delete nan values in the '2010/11/Award Amount' column. dropna command can be used to remove nan values in the columns specified. We can do this removal in place to keep the operation simpler.

##### Syntax: df.dropna(subset = ['column1_name', 'column2_name', 'column3_name'], inplace=True)

In [12]:
fundingdata.dropna(subset=['2010/11/Award Amount'],inplace=True)

#### Check the data again to see if the nan values are removed.

In [13]:
fundingdata.describe()

2010/11/Award Amount
count            757.000000
mean          420827.894320
std           287458.053183
min                1.000000
25%           162541.000000
50%           407794.000000
75%           671142.000000
max           997852.000000

#### The nan values have been removed and we are left with 757 records (instead of 831), across 46 states (instead of 50).
#### We can observe that the minimum funding is 1 USD, max funding is 997,852 USD and the mean funding is 420,828 USD.

### To conduct further analysis, we need to understand the background infomration about the funding data.
#### According to the Metadata provided for "School Improvement 2010 Grants": States were awarded nearly $3.5 billion in School #### Improvement Grant funds in 2010 to turn around their persistently lowest achieving schools. One of the following four models were to be implemented: Turnaround Model, Restart Model, School Closure and Transformation Model.
#### Only one column has numerical values and all others are strings, plotting a scatter graph may not yield much insights.
#### We can start with a bar graph for - funding by district 

In [14]:
trace1 = [go.Bar(x=fundingdata.District Name, y=fundingdata.2010/11/Award Amount)]
layout = go.Layout(title="Award Amounts by School Districts", yaxis=dict(title='Amounts'), xaxis=dict(title='School Districts'))
fig = go.Figure(data=trace1, layout=layout)
py.offline.iplot(fig)


SyntaxError: invalid syntax (<ipython-input-14-767ee9c57040>, line 1)

#### Using the default column names seem to be causing syntax errors. So lets change the names to make them easy to use

In [16]:
fundingdata.columns=['SchoolName', 'City', 'State', 'DistrictName', 'Award', 'Model', 'Address']

In [17]:
fundingdata.head()

SchoolName      City State  \
0  HOGARTH KINGEEKUK MEMORIAL SCHOOL  SAVOONGA    AK   
1                    AKIACHAK SCHOOL  AKIACHAK    AK   
2                     GAMBELL SCHOOL   GAMBELL    AK   
3               BURCHELL HIGH SCHOOL   WASILLA    AK   
4                       AKIAK SCHOOL     AKIAK    AK   

                                DistrictName     Award           Model  \
0              BERING STRAIT SCHOOL DISTRICT  471014.0  Transformation   
1                     YUPIIT SCHOOL DISTRICT  520579.0  Transformation   
2              BERING STRAIT SCHOOL DISTRICT  449592.0  Transformation   
3  MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT  641184.0  Transformation   
4                     YUPIIT SCHOOL DISTRICT  399686.0  Transformation   

                                             Address  
0  200 MAIN ST\nSAVOONGA, AK 99769\n(63.6687, -17...  
1  AKIACHAK 51100\nAKIACHAK, AK 99551\n(60.8911, ...  
2  169 MAIN ST\nGAMBELL, AK 99742\n(63.7413, -171...  
3  1775 WEST PARKS HWY\nWASILLA, AK 99654\n(61.57...  
4     AKIAK 5227\nAKIAK, AK 99552\n(60.8879, -161.2)

In [18]:
totalaward=fundingdata.Award.sum()
totalaward

318566716.0

#### Total Funding in the year 2010-201 is USD 318,566,716.

In [19]:
trace1 = [go.Bar(x=fundingdata.DistrictName,
                 y=fundingdata.Award)]
layout = go.Layout(title="Award Amounts by School Districts",
                   yaxis=dict(title='Award Amounts'),
                   xaxis=dict(title='School Districts'))
fig = go.Figure(data=trace1, layout=layout)
py.offline.iplot(fig)


#### Plot a Bar chart to show funding for each State

In [20]:
trace2 = [go.Bar(x=fundingdata.State, y=fundingdata.Award)]
layout = go.Layout(title="Award Amounts by States", yaxis=dict(title='Award Amounts'), xaxis=dict(title='States'))
fig = go.Figure(data=trace2, layout=layout)
py.offline.iplot(fig)

#### Now a Bar chart to show funding for each Model

In [21]:
trace3 = [go.Bar(x=fundingdata.Model,
                 y=fundingdata.Award)]
layout = go.Layout(title="Award Amounts by Model Types",
                   yaxis=dict(title='Award Amounts'),
                   xaxis=dict(title='Model'))
fig = go.Figure(data=trace3, layout=layout)
py.offline.iplot(fig)

#### Lets review funding for each state by grouping the amounts

In [22]:
grouping=fundingdata.groupby(['State'])[['Award']].sum()

In [23]:
grouping

Award
State            
AK      3607416.0
AL      2300782.0
AR      3936109.0
AZ     13181345.0
CA     20286824.0
CO     10158232.0
CT      8041137.0
DC      4199339.0
DE      1303216.0
FL     43138859.0
GA      8892929.0
IA       761235.0
ID       496680.0
IL      2197110.0
IN      2191767.0
KS      1270660.0
KY      5025938.0
MA      7441266.0
MD      6033577.0
MI     10893630.0
MN     11633397.0
MO     13215147.0
MS      1250153.0
MT      1961690.0
NC     13527138.0
ND       418574.0
NE      4078079.0
NH      2661604.0
NJ      2158303.0
NM      1975170.0
NV      4991568.0
NY      5668851.0
OH     15266786.0
OK      2770485.0
OR      5792028.0
PA     28318207.0
SC     11272707.0
SD       530890.0
TN      4997708.0
UT      4251625.0
VA      8008990.0
VT      2750483.0
WA      6569211.0
WI      3176096.0
WV      4961093.0
WY      1002682.0

#### Let us now separete out the grant amounts for each model

In [24]:
fundingdatatransform=fundingdata.groupby(['Model']).get_group('Transformation')
fundingdatatransform

SchoolName               City State  \
0     HOGARTH KINGEEKUK MEMORIAL SCHOOL           SAVOONGA    AK   
1                       AKIACHAK SCHOOL           AKIACHAK    AK   
2                        GAMBELL SCHOOL            GAMBELL    AK   
3                  BURCHELL HIGH SCHOOL            WASILLA    AK   
4                          AKIAK SCHOOL              AKIAK    AK   
6                       TULUKSAK SCHOOL           TULUKSAK    AK   
7           CAPITOL HEIGHTS JR HIGH SCH         MONTGOMERY    AL   
8                 LOWNDES CO MIDDLE SCH       FORT DEPOSIT    AL   
9                    CENTRAL MIDDLE SCH           ROCKFORD    AL   
10              BELLINGRATH JR HIGH SCH         MONTGOMERY    AL   
11                HAYNEVILLE MIDDLE SCH         HAYNEVILLE    AL   
12                 SOUTHLAWN MIDDLE SCH         MONTGOMERY    AL   
13              CONTINUOUS LEARNING CTR             MOBILE    AL   
14              JACKSON-STEELE ELEM SCH         HAYNEVILLE    AL   
15                  MCINTYRE MIDDLE SCH         MONTGOMERY    AL   
16                  JOHN ESSEX HIGH SCH          DEMOPOLIS    AL   
17                     CALHOUN HIGH SCH        LETOHATCHEE    AL   
18                OSCEOLA MIDDLE SCHOOL            OSCEOLA    AR   
19                     HALL HIGH SCHOOL        LITTLE ROCK    AR   
20              ROSE CITY MIDDLE SCHOOL  NORTH LITTLE ROCK    AR   
21             TRUSTY ELEMENTARY SCHOOL         FORT SMITH    AR   
22                J.A. FAIR HIGH SCHOOL        LITTLE ROCK    AR   
23                  OSCEOLA HIGH SCHOOL            OSCEOLA    AR   
24             CLOVERDALE MIDDLE SCHOOL        LITTLE ROCK    AR   
25            GADSDEN ELEMENTARY SCHOOL            GADSDEN    AZ   
28                 ALCHESAY HIGH SCHOOL         WHITERIVER    AZ   
29               SAN CARLOS HIGH SCHOOL         SAN CARLOS    AZ   
32             PROJECT MORE HIGH SCHOOL             TUCSON    AZ   
33                 PEACH SPRINGS SCHOOL      PEACH SPRINGS    AZ   
34               WHITE CONE HIGH SCHOOL         WHITE CONE    AZ   
..                                  ...                ...   ...   
793                           JASON LEE             TACOMA    WA   
794        HIGH POINT ELEMENTARY SCHOOL            SEATTLE    WA   
795            MONTICELLO MIDDLE SCHOOL           LONGVIEW    WA   
796          STANTON ALTERNATIVE SCHOOL             YAKIMA    WA   
799               CLEVELAND HIGH SCHOOL            SEATTLE    WA   
800               CHINOOK MIDDLE SCHOOL             SEATAC    WA   
801            WASHINGTON MIDDLE SCHOOL             YAKIMA    WA   
805       JAMES MADISON ACADEMIC CAMPUS          MILWAUKEE    WI   
806                        VINCENT HIGH          MILWAUKEE    WI   
807                       BAY VIEW HIGH          MILWAUKEE    WI   
809                 SOUTH DIVISION HIGH          MILWAUKEE    WI   
812             BRADLEY TECHNOLOGY HIGH          MILWAUKEE    WI   
813                         HAMLIN PK-8             HAMLIN    WV   
814               RIVERSIDE HIGH SCHOOL              BELLE    WV   
815  DODDRIDGE COUNTY ELEMENTARY SCHOOL         WEST UNION    WV   
816              MOUNT VIEW HIGH SCHOOL              WELCH    WV   
817              WEST HAMLIN ELEMENTARY        WEST HAMLIN    WV   
818            MALDEN ELEMENTARY SCHOOL         CHARLESTON    WV   
819     MARTINSBURG NORTH MIDDLE SCHOOL        MARTINSBURG    WV   
820           SANDY RIVER MIDDLE SCHOOL           AVONDALE    WV   
821     STONEWALL JACKSON MIDDLE SCHOOL         CHARLESTON    WV   
822             EAST BANK MIDDLE SCHOOL          EAST BANK    WV   
823          FRANKLIN ELEMENTARY CENTER        PARKERSBURG    WV   
824                 GUYAN VALLEY MIDDLE         BRANCHLAND    WV   
825           CEDAR GROVE MIDDLE SCHOOL        CEDAR GROVE    WV   
826              MOUNT HOPE HIGH SCHOOL         MOUNT HOPE    WV   
827             GEARY ELEMENTARY SCHOOL          LEFT HAND    WV   
828              

In [25]:
totaltransform=fundingdatatransform.Award.sum()
totaltransform

235725022.0

In [26]:
fundingdatarestart=fundingdata.groupby(['Model']).get_group('Restart')
fundingdatarestart

SchoolName                  City State  \
5                               MIDVALLEY HIGH               WASILLA    AK   
52         FLORENCE GRIFFITH JOYNER ELEMENTARY           LOS ANGELES    CA   
57                       SAMUEL GOMPERS MIDDLE           LOS ANGELES    CA   
74                        EDWIN MARKHAM MIDDLE           LOS ANGELES    CA   
107            GEORGE WASHINGTON CARVER MIDDLE           LOS ANGELES    CA   
119              ROBERT LOUIS STEVENSON MIDDLE           LOS ANGELES    CA   
141                         LAKE MIDDLE SCHOOL                DENVER    CO   
162                        HARDING HIGH SCHOOL            BRIDGEPORT    CT   
169                              ANACOSTIA SHS  DISTRICT OF COLUMBIA    DC   
171                                 DUNBAR SHS  DISTRICT OF COLUMBIA    DC   
178                              KENILWORTH ES  DISTRICT OF COLUMBIA    DC   
299               PHILLIPS ACADEMY HIGH SCHOOL               CHICAGO    IL   
338                    WILLIAM C. MARCH MIDDLE             BALTIMORE    MD   
339                          CHINQUAPIN MIDDLE             BALTIMORE    MD   
342                                  CALVERTON             BALTIMORE    MD   
344                            GARRISON MIDDLE             BALTIMORE    MD   
347           COMMODORE JOHN ROGERS ELEMENTARY             BALTIMORE    MD   
467                    ANSON CHALLENGE ACADEMY             WADESBORO    NC   
488                        RENAISSANCE ACADEMY                NEWARK    NJ   
606                      DOUGLAS STEPHEN A SCH          PHILADELPHIA    PA   
621                       SMEDLEY FRANKLIN SCH          PHILADELPHIA    PA   
625                      HARRITY WILLIAM F SCH          PHILADELPHIA    PA   
646                          STETSON JOHN B MS          PHILADELPHIA    PA   
648                          DAROFF SAMUEL SCH          PHILADELPHIA    PA   
649                       BLUFORD GUION EL SCH          PHILADELPHIA    PA   
652                         MANN WILLIAM B SCH          PHILADELPHIA    PA   
763                      LANGSTON FOCUS SCHOOL              DANVILLE    VA   
764                             ARMSTRONG HIGH              RICHMOND    VA   
768                   ELLEN W. CHAMBLISS ELEM.             WAKEFIELD    VA   
769                  THOMAS C. BOUSHALL MIDDLE              RICHMOND    VA   
773                      SUSSEX CENTRAL MIDDLE                SUSSEX    VA   
804                                CUSTER HIGH             MILWAUKEE    WI   
810  MILWAUKEE AFRICAN AMERICAN IMMERSION HIGH             MILWAUKEE    WI   

                                          DistrictName     Award    Model  \
5            MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT  697703.0  Restart   
52                                 LOS ANGELES UNIFIED  188694.0  Restart   
57                                 LOS ANGELES UNIFIED  189201.0  Restart   
74                                 LOS ANGELES UNIFIED  185886.0  Restart   
107                                LOS ANGELES UNIFIED  174824.0  Restart   
119                                LOS ANGELES UNIFIED  189229.0  Restart   
141  SCHOOL DISTRICT NO. 1 IN THE COUNTY OF DENVER ...  619609.0  Restart   
162                         BRIDGEPORT SCHOOL DISTRICT  800000.0  Restart   
169                DISTRICT OF COLUMBIA PUBLIC SCHOOLS  517340.0  Restart   
171                DISTRICT OF COLUMBIA PUBLIC SCHOOLS  517340.0  Restart   
178                DISTRICT OF COLUMBIA PUBLIC SCHOOLS  305325.0  Restart   
299                             CITY OF CHICAGO SD 299  179737.0  Restart   
338                      BALTIMORE CITY PUBLIC SCHOOLS  956457.0  Restart   
339                      BALTIMORE CITY PUBLIC SCHOOLS  465091.0  Restart   
342                      BALTIMORE CITY PUBLIC SCHOOLS  177759.0  Restart   
344                      BALTIMORE CITY PUBLIC SCHOOLS  120707.0  Restart   
347                      BALTIMORE CITY PUBLIC SCHOOLS  102706.0  Restart 

In [27]:
totalrestart=fundingdatarestart.Award.sum()
totalrestart

16612610.0

In [28]:
fundingdataturn=fundingdata.groupby(['Model']).get_group('Turnaround')
fundingdataturn

SchoolName            City State  \
26                             EAST VALLEY HIGH SCHOOL            MESA    AZ   
27                            CRITTENTON YOUTH ACADEMY         PHOENIX    AZ   
30                        PIMA PARTNERSHIP SCHOOL  THE          TUCSON    AZ   
31                               LA PUERTA HIGH SCHOOL         PHOENIX    AZ   
39                             BABOQUIVARI HIGH SCHOOL           SELLS    AZ   
40                     IMAGINE ELEMENTARY AT CAMELBACK         PHOENIX    AZ   
43                                SELIGMAN HIGH SCHOOL        SELIGMAN    AZ   
46                                 LONGWOOD ELEMENTARY         HAYWARD    CA   
50                               TUMBLEWEED ELEMENTARY        PALMDALE    CA   
63                   EDISON-RONALD MCNAIR INTERMEDIATE  EAST PALO ALTO    CA   
68                                        SEASIDE HIGH         SEASIDE    CA   
69                                     ELLA ELEMENTARY      OLIVEHURST    CA   
70                                   WILSON ELEMENTARY  SAN BERNARDINO    CA   
71                                     YOSEMITE MIDDLE          FRESNO    CA   
73                             ENCINA PREPARATORY HIGH      SACRAMENTO    CA   
76                                 FELICITA ELEMENTARY       ESCONDIDO    CA   
80                                   BARTON ELEMENTARY  SAN BERNARDINO    CA   
81                             ROSE FERRERO ELEMENTARY         SOLEDAD    CA   
82                 GEORGE WASHINGTON CARVER ELEMENTARY   SAN FRANCISCO    CA   
85                                  COSTANO ELEMENTARY  EAST PALO ALTO    CA   
90                                  MARTIN LUTHER KING         SEASIDE    CA   
95                                SHANDIN HILLS MIDDLE  SAN BERNARDINO    CA   
96                                  WEBSTER ELEMENTARY          FRESNO    CA   
98                                      EVERETT MIDDLE   SAN FRANCISCO    CA   
100                                MARSHALL ELEMENTARY  SAN BERNARDINO    CA   
102                                   WEST SHORES HIGH     SALTON CITY    CA   
104                               JOHN MUIR ELEMENTARY   SAN FRANCISCO    CA   
105                                 LINCOLN ELEMENTARY        RICHMOND    CA   
108                                  BRYANT ELEMENTARY   SAN FRANCISCO    CA   
114                                HIGHLAND ELEMENTARY         SEASIDE    CA   
..                                                 ...             ...   ...   
525  Robert Brown High School of Construction and D...       Rochester    NY   
531                              EMERSON MIDDLE SCHOOL         YONKERS    NY   
532                    Vanguard Collegiate High School       Rochester    NY   
534                   GLOBAL MEDIA ARTS HS AT FRANKLIN       ROCHESTER    NY   
543                         Rochester STEM High School       Rochester    NY   
546                              WOODLAND HILLS SCHOOL       CLEVELAND    OH   
549                              FRANKLIN D. ROOSEVELT       CLEVELAND    OH   
551                              GLENVILLE HIGH SCHOOL       CLEVELAND    OH   
557                           LINCOLN-WEST HIGH SCHOOL       CLEVELAND    OH   
559                         EAST TECHNICAL HIGH SCHOOL       CLEVELAND    OH   
568                             COLLINWOOD HIGH SCHOOL       CLEVELAND    OH   
574                               MARY B MARTIN SCHOOL       CLEVELAND    OH   
576                         JOHN F KENNEDY HIGH SCHOOL       CLEVELAND    OH   
590                                     U. S. GRANT HS             OKC    OK   
611                              STEELTON-HIGHSPIRE HS        STEELTON    PA   
617                                            HAND MS       LANCASTER    PA   
620                                 UNIVERSITY CITY HS    PHILADELPHIA    PA   
631                                    VAUX ROBERTS HS    PHILADELPHIA    PA   
633                                CLEMENTE 

In [29]:
totalturn=fundingdataturn.Award.sum()
totalturn

65026596.0

In [30]:
fundingdataclose=fundingdata.groupby(['Model']).get_group('Closure')
fundingdataclose

SchoolName           City State  \
103                                    EXPLORE MIDDLE        OAKLAND    CA   
133                BROWN, JR., (WILLIE L.) ELEMENTARY  SAN FRANCISCO    CA   
145                              RISHEL MIDDLE SCHOOL         DENVER    CO   
149                     SKYLAND COMMUNITY HIGH SCHOOL         DENVER    CO   
150                         PHILIPS ELEMENTARY SCHOOL         DENVER    CO   
428                                    ERVIN JR. HIGH    KANSAS CITY    MO   
615                                ROONEY ARTHUR J MS     PITTSBURGH    PA   
632                         CAREER TECHNOLOGY ACADEMY     HARRISBURG    PA   
675                                  WHITLOCK JR HIGH    SPARTANBURG    SC   
766                                 NEW BRIDGE SCHOOL       RICHMOND    VA   
771                           VA. RANDOLPH COMM. HIGH     GLEN ALLEN    VA   
798                                              HUNT         TACOMA    WA   
802  FOSTER & WILLIAMS HIGH SCHOOL OF THE VISUAL ARTS      MILWAUKEE    WI   
803                                       DUBOIS HIGH      MILWAUKEE    WI   
808       DOWNTOWN INSTITUTE OF ARTS & LETTERS (DIAL)      MILWAUKEE    WI   
811                WHS LAW EDUCATION & PUBLIC SERVICE      MILWAUKEE    WI   

                                          DistrictName     Award    Model  \
103                                    OAKLAND UNIFIED   49045.0  Closure   
133                              SAN FRANCISCO UNIFIED   50000.0  Closure   
145  SCHOOL DISTRICT NO. 1 IN THE COUNTY OF DENVER ...   13729.0  Closure   
149  SCHOOL DISTRICT NO. 1 IN THE COUNTY OF DENVER ...   31935.0  Closure   
150  SCHOOL DISTRICT NO. 1 IN THE COUNTY OF DENVER ...   32491.0  Closure   
428                                  HICKMAN MILLS C-1   46872.0  Closure   
615                                      PITTSBURGH SD  208773.0  Closure   
632                                 HARRISBURG CITY SD  300000.0  Closure   
675                                     SPARTANBURG 07  254323.0  Closure   
766                               HENRICO CO PBLC SCHS   50000.0  Closure   
771                               HENRICO CO PBLC SCHS   50000.0  Closure   
798                             TACOMA SCHOOL DISTRICT   95320.0  Closure   
802                          MILWAUKEE SCHOOL DISTRICT    5000.0  Closure   
803                          MILWAUKEE SCHOOL DISTRICT    5000.0  Closure   
808                          MILWAUKEE SCHOOL DISTRICT    5000.0  Closure   
811                          MILWAUKEE SCHOOL DISTRICT    5000.0  Closure   

                                               Address  
103  3550 64TH AVE\nOAKLAND, CA 94605\n(37.7757, -1...  
133  2055 SILVER AVE\nSAN FRANCISCO, CA 94124\n(37....  
145  451 SOUTH TEJON ST\nDENVER, CO 80223\n(39.7084...  
149  3240 HUMBOLDT ST\nDENVER, CO 80205\n(39.7632, ...  
150  6550 EAST 21ST AVE\nDENVER, CO 80207\n(39.7486...  
428  10530 GREENWOOD RD\nKANSAS CITY, MO 64134\n(38...  
615  3530 FLEMING AVE\nPITTSBURGH, PA 15212\n(40.47...  
632  2915 NORTH THIRD ST\nHARRISBURG, PA 17110\n(40...  
675  364 SUCCESSFUL WAY\nSPARTANBURG, SC 29303\n(34...  
766  5915 NINE MILE RD\nRICHMOND, VA 23223\n(37.551...  
771  2204 MOUNTAIN RD\nGLEN ALLEN, VA 23060\n(37.66...  
798  6501 SOUTH 10TH ST\nTACOMA, WA 98465\n(47.2527...  
802  6700 NORTH 80TH ST\nMILWAUKEE, WI 53223\n(43.1...  
803  4141 NORTH 64TH ST\nMILWAUKEE, WI 53216\n(43.0...  
808  227 WEST PLACEEASANT ST\nMILWAUKEE, WI 53212\n...  
811  2525 NORTH SHERMAN BLVD\nMILWAUKEE, WI 53210\n...

In [31]:
totalclose=fundingdataclose.Award.sum()
totalclose

1202488.0

#### Using "Figure Factory" create a table of School Names, City, State, Award, and Model Type - Closure

In [32]:
closuretable = ff.create_table(fundingdataclose[["SchoolName", "City", "Award", "Model"]])
py.offline.iplot(closuretable)

#### Using "Figure Factory" create a table of School Names, City, State, Award, and Model Type - Turn Around

In [33]:
turntable = ff.create_table(fundingdataturn[["SchoolName", "City", "Award", "Model"]])
py.offline.iplot(turntable)